File: examples/thrift/thrift_example.py
import sys,os
gen_py_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'lib', 'proto', 'gen-py')
sys.path.append(gen_py_path)

from cloudthrift import CommandInterface
from cloudthrift.constants import *

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

transport = TSocket.TSocket('localhost', 11223)
transport = TTransport.TBufferedTransport(transport)
protocol = TBinaryProtocol.TBinaryProtocol(transport)

ci = CommandInterface.Client(protocol)

transport.open()

cloudquery = CloudQuery()
cloudquery.name = "app"

resp = ci.run_command(cloudquery, "name", "")
print resp.response

resp = ci.run_command(cloudquery, "maximum_instances", "")
print resp.response

File: examples/thrift/thrift_example.rb
#!/usr/bin/env ruby

$:.unshift(File.dirname(__FILE__) + "/../../lib")
$:.unshift(File.dirname(__FILE__) + "/../../examples")
require "poolparty"
require "simple"

$:.unshift("#{File.dirname(__FILE__)}/../../lib/proto/gen-rb")

require "thrift"
require "command_interface"
require "poolparty_constants"
require "poolparty_types"

port = ARGV.pop || 11223

transport = Thrift::BufferedTransport.new(Thrift::Socket.new('localhost', port))
protocol = Thrift::BinaryProtocol.new(transport)

client = CloudThrift::CommandInterface::Client.new(protocol)
transport.open()

cld = CloudThrift::CloudQuery.new
cld.name = 'monitored_app'

resp = client.run_command(cld, "name", [])
puts resp.response

resp = client.run_command(cld, "maximum_instances", [])
puts resp.response

resp = client.run_command(cld, "run_monitor", ["cpu", "0.91"])
p resp.response

resp = client.run_command(cld, "run_monitor", ["cpu", "0.01"])
p resp.response